### PR TITLE
Fixed registration date/block discrepancy

### PR DESCRIPTION
**Pull request by shahafn**

### CODE REVIEW

File: server/src/librelay/relay_server.go [hunk 1]
```diff
@@ -324,38 +324,35 @@ func (relay *RelayServer) IsUnstaked() (removed bool, err error) {
 	return true, nil
 }
 
-func (relay *RelayServer) RegistrationDate() (when int64, err error) {
+func (relay *RelayServer) BlockCountSinceRegistration() (when uint64, err error) {
```
Comment (drortirosh) [inline server/src/librelay/relay_server.go]: Block count is not a `when`, but `count`

File: server/src/librelay/relay_server.go [hunk 2]
```diff
@@ -324,38 +324,35 @@ func (relay *RelayServer) IsUnstaked() (removed bool, err error) {
...
 	if err != nil {
 		log.Println(err)
 		return
 	}
-
+	// We only care about the last registration event
+	for iter.Next() {}
 	if (iter.Event == nil && !iter.Next()) ||
 		(bytes.Compare(iter.Event.Relay.Bytes(), relay.Address().Bytes()) != 0) ||
 		(iter.Event.TransactionFee.Cmp(relay.Fee) != 0) ||
```
Comment (drortirosh) [inline server/src/librelay/relay_server.go]: If you check the txfee when checking for the last event, then it means a relay can't change it's txfee... Remove this check, as it adds nothing (you can argue whether a check for the url is required. I think it isn't, but at least, it's not harmful either, but txfee shouldn't be checked)

File: server/src/relay/RelayHttpServer.go [hunk 3]
```diff
@@ -178,7 +178,7 @@ func parseCommandLine() (relayParams librelay.RelayParams) {
 	relayHubAddress := flag.String("RelayHubAddress", "0x537F27a04470242ff6b2c3ad247A05248d0d27CE", "RelayHub address")
 	defaultGasPrice := flag.Int64("DefaultGasPrice", int64(params.GWei), "Relay's default gasPrice per (non-relayed) transaction in wei")
 	gasPricePercent := flag.Int64("GasPricePercent", 10, "Relay's gas price increase as percentage from current average. GasPrice = (100+GasPricePercent)/100 * eth_gasPrice() ")
-	registrationBlockRate := flag.Uint64("RegistrationBlockRate", 5800, "Relay registeration rate (in blocks)")
+	RegistrationBlockRate = *flag.Uint64("RegistrationBlockRate", 5800, "Relay registeration rate (in blocks)")
```
Comment (drortirosh) [inline server/src/relay/RelayHttpServer.go]: Please put 6000-200 here, to reference the value the client checks.

**Comment by drortirosh** [comment]
one thing I don't see is reducing the interval between checks

**Comment by shahafn** [comment]
I did update it

File: server/src/librelay/relay_server.go (hunk 2 above)
Comment (shahafn) [reply server/src/librelay/relay_server.go]: I actually thought that if you change the tx fee then it makes sense to verify it in the event, why does it mean that you can't change it?

File: package.json [hunk 4]
```diff
@@ -33,7 +33,7 @@
   "scripts": {
     "test": "npm run test-server && npm run test-js",
     "test-server": "make test-server",
-    "test-js": "run-with-testrpc --port 8544 --gasLimit 8000000 --defaultBalanceEther 1000 --deterministic 'npx truffle --network npmtest test'",
+    "test-js": "make build-server && run-with-testrpc --port 8544 --gasLimit 8000000 --defaultBalanceEther 1000 --deterministic 'npx truffle --network npmtest test'",
```
[the remark this answers is not in the document]
Comment (shahafn) [reply package.json]: Fixed

**Pull request merged**